src/main/java/frc/robot/RobotContainer.java

```diff
@@ -388,7 +389,8 @@ private void configureBindings() {
             driveTrain.runOnce(
                 () -> driveTrain.resetPose(new Pose2d(new Translation2d(0, 0), new Rotation2d()))));
 
-    joystick.x().onTrue(new ZeroElevator(elevatorSubsystem));
+    joystick.x().onTrue(CoralPosition.isCoralInTootsieSlide() ? new EjectCoralThenZero(elevatorSubsystem, tootsieSlideSubsystem) : new ZeroElevator(elevatorSubsystem));
```

Comment:
This won't do what you expect, since it will check the state of the coral when the command is bound (on robot boot).